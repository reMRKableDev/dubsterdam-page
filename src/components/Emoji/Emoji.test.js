import React from "react";
import Emoji from "./Emoji";
import { shallow } from "enzyme";

import {
  validateMatchSnapshot,
  validateEmptyString,
  validateEmojiValue,
} from "../../utils/testValidators";

const shallowComponent = () => shallow(<Emoji />);
const shallowComponentWithProps = () =>
  shallow(<Emoji symbol="❤️" label="love" />);

describe("Emoji component unit tests", () => {
  it("should shallow render an Emoji component without props ", () => {
    const wrapper = shallowComponent();

    validateMatchSnapshot(wrapper);
    validateEmptyString(wrapper.find("span").text());
  });

  it("should shallow render an Emoji component with props ", () => {
    const wrapper = shallowComponentWithProps();
    
    validateMatchSnapshot(wrapper);
    validateEmojiValue(wrapper.find("span").text());
  });
});
