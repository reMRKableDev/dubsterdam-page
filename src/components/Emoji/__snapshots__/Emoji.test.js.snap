// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Emoji component unit tests should shallow render an Emoji component with props  1`] = `
<span
  aria-hidden="false"
  aria-label="love"
  className="emoji"
  role="img"
>
  ❤️
</span>
`;

exports[`Emoji component unit tests should shallow render an Emoji component without props  1`] = `
<span
  aria-hidden="true"
  aria-label=""
  className="emoji"
  role="img"
/>
`;
